### PR TITLE
only fail components using on with lifecylcle hooks

### PR DESCRIPTION
**Pull request by patience-tema-baron**
Resolves https://github.com/ember-cli/eslint-plugin-ember/issues/223. Let me know if you'd like any changes!

### CODE REVIEW

**Comment by Alonski** [comment]
@patience-tema-baron This is really awesome!

**Comment by btecu** [comment]
Can we get this merged?

**Comment by bgentry** [comment]
I ran into #223 today and would really love to see this PR merged too :pray:

Thanks for putting it together @patience-tema-baron

File: lib/rules/no-on-calls-in-components.js [hunk 1]
```diff
@@ -4,29 +4,43 @@ const ember = require('../utils/ember');
 const utils = require('../utils/utils');
 
 //----------------------------------------------
-// General rule - Don\'t use .on() in components
+// General rule - Don\'t use .on() to call lifecycle hooks in components
```
Comment (rwjblue) [inline lib/rules/no-on-calls-in-components.js]: Small verbiage tweak here (and throughout the rest of the file):

> Don’t use `.on()` for component lifecycle events.

**Comment by gabrielgrant** [comment]
@patience-tema-baron do you have a few minutes to make these changes? otherwise, happy to do it on a new branch

**Comment by patience-tema-baron** [comment]
Sorry about the delay, I've been on vacation. I'll be getting back over the
weekend and can do this on Tuesday, but feel free to make the changes if
you need this sooner.

Either way, sorry to be so slow about this!

**Comment by gabrielgrant** [comment]
@patience-tema-baron no worries, thanks for circling back around to this

**Comment by gabrielgrant** [comment]
@patience-tema-baron thanks for getting those changes in!

@rwjblue when you have a sec, looks like this is ready for a re-review/merge

**Comment by rwjblue** [comment]
Thank you @patience-tema-baron!

**Comment by crhayes** [comment]
Very much needed, thanks for the update! @rwjblue any insight into when the next release will be?

**Pull request merged**